### DISCUSSION TITLE
API pagination's duplicates

### DISCUSSION THREAD

**Discussion by NoeGRANGE** [plausible/analytics#2415] [Anything Else]
Hello,

I'm using your API to make my own dashboard but I think I got a problem when I try to request breakdown with `limit` and `page` parameters.
Every result from the second page already exist on the first one.

- `https://plausible.io/api/v1/stats/breakdown?site_id=esf-meribel.com&period=custom&date=2022-09-13,2022-11-02&metrics=visitors&property=event:props:orderId&filters=event:name==MseM: Payment success&limit=1000&page=1`

- `https://plausible.io/api/v1/stats/breakdown?site_id=esf-meribel.com&period=custom&date=2022-09-13,2022-11-02&metrics=visitors&property=event:props:orderId&filters=event:name==MseM: Payment success&limit=1000&page=2`

Am I using these parameters wrong?

Thanks in advance 👍

**Comment by bogplau** [comment]
Hi @NoeGRANGE 

Sounds like this is the same as #2255 

Basically, if a request returns less than the limit you requested it means that all pages have been returned so there's no need to make the second request

**Reply by NoeGRANGE** [reply]
Hi @bogplau 

Thanks but what's strange is that if I chunk my request by week I obtain more results.
That's why I think my problem's different than <a href={https://github.com/plausible/analytics/issues/2255}>#2255</a>

Thanks anyway 👍

**Reply by ukutaht** [reply]
@NoeGRANGE This requires some deeper investigation. We'll look into it and update this thread with the results. Thanks for reporting the issue!

**Reply by NoeGRANGE** [reply]
Thanks for your time @ukutaht

**Comment by vinibrsl** [comment]
Hey @NoeGRANGE! We just released a [patch](https://github.com/plausible/analytics/pull/2562) to fix the pagination issues with the Breakdown API. Could you give it a quick test and let us know if the issue is fixed for you? Thanks for reporting!
